### PR TITLE
Add incident support.

### PR DESCRIPTION
**Pull request by r0xsh**
Fixes: #1755

### CODE REVIEW

File: src/redux/Courier/taskEntityReducer.js [hunk 1]
```diff
@@ -81,6 +84,17 @@ function replaceItem(state, payload) {
   return state;
 }
 
+
+function updateItem(state, id, payload) {
+  const index = state.findIndex(item => item['@id'] === id);
+
+  if (index !== -1) {
+    return state.map((item, i) => (i === index ? { ...item, ...payload } : item));
```
Comment (vladimir-8) [inline src/redux/Courier/taskEntityReducer.js:92]: why `map`? can be just `state[index] = { ...state[index], ...payload };` ?

File: src/redux/Courier/taskEntityReducer.js [hunk 2]
```diff
@@ -81,6 +84,17 @@ function replaceItem(state, payload) {
   return state;
 }
 
+
+function updateItem(state, id, payload) {
```
Comment (vladimir-8) [inline src/redux/Courier/taskEntityReducer.js:88]: I'd call it something like `prevItems` instead of `state`

**Review by vladimir-8 — APPROVED** [review]

**Comment by vladimir-8** [comment]
I guess we will need to wait with merging this PR until all instances are updated? Otherwise we will not be able to release the app, as it relies on newer APIs

File: src/redux/Courier/taskActions.js [hunk 3]
```diff
@@ -28,6 +28,9 @@ export const MARK_TASK_FAILED_FAILURE = 'MARK_TASK_FAILED_FAILURE';
 export const START_TASK_REQUEST = 'START_TASK_REQUEST';
 export const START_TASK_SUCCESS = 'START_TASK_SUCCESS';
 export const START_TASK_FAILURE = 'START_TASK_FAILURE';
+export const REPORT_INCIDENT_REQUEST = 'REPORT_INCIDENT_REQUEST';
```
[the remark this answers is not in the document]
Comment (r0xsh) [reply src/redux/Courier/taskActions.js:32]: Could be in its own PR, where we refactor all the actions.

File: src/redux/Courier/taskEntityReducer.js (hunk 1 above)
Comment (r0xsh) [reply src/redux/Courier/taskEntityReducer.js:92]: For immutability, objects in javascript are passed via reference. So i should not edit the `state` variable.

Comment (r0xsh) [reply src/redux/Courier/taskEntityReducer.js:92]: You'd to clone the object in order to use your snippet
```javascript
const updatedItems = [...prevItems]; // Create a copy of the array
updatedItems[index] = { ...updatedItems[index], ...payload };
```

It's possible that way, but i'm more used to functional paradigm

**Pull request merged**